lib/astutils.cpp

```diff
@@ -1131,6 +1131,35 @@ bool isConstVarExpression(const Token *tok)
     return false;
 }
 
+static const Variable *getLHSVariableRecursive(const Token *tok)
+{
+    if (!tok)
+        return nullptr;
+    if (Token::Match(tok, "*|&|&&|[")) {
+        const Variable *var1 = getLHSVariableRecursive(tok->astOperand1());
+        if (var1 || Token::simpleMatch(tok, "["))
+            return var1;
+        const Variable *var2 = getLHSVariableRecursive(tok->astOperand2());
+        return var2;
+    }
+    if (Token::Match(tok->previous(), "this . %var%"))
+        return getLHSVariableRecursive(tok->next());
+    if (!tok->variable())
```

Comment:
This `if` looks redundant. It seems you can return `tok->variable()` unconditionally here.